components/ca-form.js

```diff
@@ -0,0 +1,1520 @@
+
+                            default: {
+
+                                formData[key] = (element.value || '').trim();
+
+                                if (item.format && item.format === 'currency' && !formData.$currency) {
+                                    formData.$currency = componentSupport.getUserCurrency().code;
+                                }
+                            }
+                        }
+                    }
+                }
+            }
+
+            return formData;
+        }
+
+        /**
+         * @description clears validation errors
+         * @returns {void} nothing
+        */
+        clearValidationErrors() {
+
+            // clear previous validation errors
+            this.form.removeAttribute('data-error');
+
+            Array.prototype.slice.call(this.querySelectorAll('[data-error]')).forEach(item =>
+                item.removeAttribute('data-error')
+            );
+        }
+
+        /**
+         * @description sets an error
+         * @param {string} fieldName name of field to set error on
+         * @param {error} error passed
+         * @returns {void} nothing
+        */
+        setError(fieldName, error) {
+
+            if (error !== '') {
+
+                const el = this.querySelector(`[name="${fieldName}"]`);
+
+                if (el) {
+                    // mark field as invalid
+                    el.setAttribute('data-invalid', 'true');
+                    el.parentNode.setAttribute('data-error', error);
+                }
+            }
+        }
+
+        /**
+         * @description clears an error
+         * @param {string} fieldName name of field to set error on
+         * @returns {void} nothing
+        */
+        clearError(fieldName) {
+
+            const el = this.querySelector(`[name="${fieldName}"]`);
+
+            if (el) {
+                el.removeAttribute('data-invalid');
+                el.parentNode.removeAttribute('data-error');
+            }
+        }
+
+        /**
+         * @description Executes validation for an individual field
+         * @param {string} key, field in schema data
+         * @param {string} value, value from the item
+         * @returns {void} nothing
+         */
+        validateField(key, value) {
+
+            // create fake data object
+            const dataItem = {};
+
+            // populate key
+            dataItem[key] = value;
+
+            // execute regular form validation but passing a single propety to test
+            this.isValid(dataItem);
+        }
+
+        /**
+         * @description Returns true if form data is valid, otherwise false. Also updates the UI with any validation errors
+         * @param {Object} data, form data
+         * @returns {Boolean} if it's a valid object or not
+         */
+        isValid(data) {
+
+            // if validation has been disabled (for example, the Form Builder doesn't want/need it)
+            if (this.disableValidation === true) {
```

Comment:
it seems this.disableValidation is always true or false... So perhaps it's a little over the top. I'll leave it for now though.